src/client/Microsoft.Identity.Client.Desktop/WamInterop.cs

```diff
@@ -14,20 +14,20 @@ namespace Microsoft.Identity.Client.Platforms
 {
     internal static class WebAuthenticationCoreManagerInterop
     {
-        public static IAsyncOperation<WebTokenRequestResult> RequestTokenForWindowAsync(IntPtr hWnd, WebTokenRequest request)
```

Comment:
These operations are async, but not in the traditional "way", but in the Windows way. You can create your own async API which can be await-ed by simply implementing an interface. The "async" suffix should be here. Please disable the rule for this file.